test/functorch/test_aotdispatch.py

```diff
@@ -2908,6 +2940,29 @@ def false_fn(x):
         inp = torch.randn(2, 2)
         gm, _ = aot_export_module(M(), [inp], trace_joint=False, pre_dispatch=True)
 
+    def test_aot_export_predispatch_conv_and_bn(self):
+        class ConvBatchnorm(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.conv = torch.nn.Conv2d(1, 3, 1, 1)
+                self.bn = torch.nn.BatchNorm2d(3)
+
+            def forward(self, x):
+                x = self.conv(x)
+                x = self.bn(x)
+                return (x,)
+
+        mod = ConvBatchnorm()
+        mod.train()
+        inp = torch.randn(1, 1, 3, 3)
+
+        gm, _ = aot_export_module(mod, [inp], trace_joint=False, pre_dispatch=True)
+        self.assertExpectedInline(str(gm.code).strip(), """\
+def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1, arg6_1, arg7_1):
+    conv2d = torch.ops.aten.conv2d.default(arg7_1, arg0_1, arg1_1);  arg7_1 = arg0_1 = arg1_1 = None
+    add = torch.ops.aten.add.Tensor(arg6_1, 1);  arg6_1 = None
+    batch_norm = torch.ops.aten.batch_norm.default(conv2d, arg2_1, arg3_1, arg4_1, arg5_1, True, 0.1, 1e-05, True);  conv2d = arg2_1 = arg3_1 = None
```

Comment:
batch_norm / native_batch_norm are two examples: we can't blindly send them to the functionalization fallback, because they sometimes-mutate their inputs (during training). So we will incorrectly capture a graph in this case, that advertises as "never mutating"

Reply:
Are you saying torch.ops.aten.batch_norm.default is actually the mutating version?